Foundation/NSURLRequest.swift

```diff
@@ -348,7 +385,12 @@ public class NSMutableURLRequest : NSURLRequest {
         @param value the header field value. 
         @param field the header field name (case-insensitive). 
     */
-    public func setValue(value: String?, forHTTPHeaderField field: String) { NSUnimplemented() }
+    public func setValue(value: String?, forHTTPHeaderField field: String) {
+        if _httpHeaderFields == nil {
+            _httpHeaderFields = [:]
+        }
+        _httpHeaderFields?[field.lowercaseString] = value
```

Comment:
The Obj-C Foundation framework preserves the capitalization as entered (as seen in what is returned from `allHTTPHeaderFields`), though is otherwise indeed case insensitive. Keeping the capitalization as is helps with readability during debugging network calls.